diff --git a/src/drivers/evdev/lv_evdev.c b/src/drivers/evdev/lv_evdev.c
index 1b8319e04..87a40ae0d 100644
--- a/src/drivers/evdev/lv_evdev.c
+++ b/src/drivers/evdev/lv_evdev.c
@@ -91,38 +91,42 @@ struct _lv_evdev_discovery_t {
  *   STATIC FUNCTIONS
  **********************/
 
-static int _evdev_process_key(uint16_t code)
-{
-    switch(code) {
-        case KEY_UP:
-            return LV_KEY_UP;
-        case KEY_DOWN:
-            return LV_KEY_DOWN;
-        case KEY_RIGHT:
-            return LV_KEY_RIGHT;
-        case KEY_LEFT:
-            return LV_KEY_LEFT;
-        case KEY_ESC:
-            return LV_KEY_ESC;
-        case KEY_DELETE:
-            return LV_KEY_DEL;
-        case KEY_BACKSPACE:
-            return LV_KEY_BACKSPACE;
-        case KEY_ENTER:
-            return LV_KEY_ENTER;
-        case KEY_NEXT:
-        case KEY_TAB:
-            return LV_KEY_NEXT;
-        case KEY_PREVIOUS:
-            return LV_KEY_PREV;
-        case KEY_HOME:
-            return LV_KEY_HOME;
-        case KEY_END:
-            return LV_KEY_END;
-        default:
-            return 0;
-    }
-}
+ static int _evdev_process_key(uint16_t code)
+ {
+     switch(code) {
+         case KEY_UP:
+             return LV_KEY_UP;
+         case KEY_DOWN:
+             return LV_KEY_DOWN;
+         case KEY_RIGHT:
+             return LV_KEY_RIGHT;
+         case KEY_LEFT:
+             return LV_KEY_LEFT;
+         case KEY_ESC:
+         case KEY_4:
+             return LV_KEY_ESC;
+         case KEY_DELETE:
+             return LV_KEY_DEL;
+         case KEY_BACKSPACE:
+             return LV_KEY_BACKSPACE;
+         case KEY_ENTER:
+         case KEY_3:
+             return LV_KEY_ENTER;
+         case KEY_NEXT:
+         case KEY_TAB:
+         case KEY_2:
+             return LV_KEY_NEXT;
+         case KEY_PREVIOUS:
+         case KEY_1:
+             return LV_KEY_PREV;
+         case KEY_HOME:
+             return LV_KEY_HOME;
+         case KEY_END:
+             return LV_KEY_END;
+         default:
+             return 0;
+     }
+ }
 
 static int _evdev_calibrate(int v, int in_min, int in_max, int out_min, int out_max)
 {
